lib/ood_core/batch_connect/templates/vnccontainer.rb

```diff
@@ -0,0 +1,248 @@
+        # @option context [#to_s] :websockify_cmd
+        #   ("${WEBSOCKIFY_CMD:-/opt/websockify/run}") the path to the
+        #   websockify script (assumes you don't modify `:after_script`)
+        # @option context [#to_s] :vnc_log ("vnc.log") path to vnc server log
+        #   file (assumes you don't modify `:before_script` or `:after_script`)
+        # @option context [#to_s] :vnc_passwd ("vnc.passwd") path to the file
+        #   generated that contains the encrypted vnc password (assumes you
+        #   don't modify `:before_script`)
+        # @option context [#to_s] :vnc_args arguments used when starting up the
+        #   vnc server (overrides any specific vnc argument) (assumes you don't
+        #   modify `:before_script`)
+        # @option context [#to_s] :name ("") name of the vnc server session
+        #   (not set if blank or `:vnc_args` is set) (assumes you don't modify
+        #   `:before_script`)
+        # @option context [#to_s] :geometry ("") resolution of vnc display (not
+        #   set if blank or `:vnc_args` is set) (assumes you don't modify
+        #   `:before_script`)
+        # @option context [#to_s] :dpi ("") dpi of vnc display (not set if
+        #   blank or `:vnc_args` is set) (assumes you don't modify
+        #   `:before_script`)
+        # @option context [#to_s] :fonts ("") command delimited list of fonts
+        #   available in vnc display (not set if blank or `:vnc_args` is set)
+        #   (assumes you don't modify `:before_script`)
+        # @option context [#to_s] :idle ("") timeout vnc server if no
+        #   connection in this amount of time in seconds (not set if blank or
+        #   `:vnc_args` is set) (assumes you don't modify `:before_script`)
+        # @option context [#to_s] :extra_args ("") any extra arguments used
+        #   when initializing the vnc server process (not set if blank or
+        #   `:vnc_args` is set) (assumes you don't modify `:before_script`)
+        # @option context [#to_s] :vnc_clean ("...") script used to clean up
+        #   any active vnc sessions (assumes you don't modify `:before_script`
+        #   or `:clean_script`)
+        # @option context [#to_s] :container_path ("vnc_container.sif") the path
+        #   to the container with VNC
+        # @option context [#to_s] :container_bindpath ("") paths to bind into
+        #   the container with VNC
+        # @option context [#to_s] :container_module ("singularity") the module
+        #   that loads Singularity or Apptainer with Lmod. Supports versions (i.e.
+        #   apptainer/1.10). If Singularity or Apptainer are installed at a 
+        #   system level (i.e., no module loaded to activate), set this to an
+        #   empty string.
+        # @option context [#to_s] :container_command ("singularity") the 
+        #   singularity or apptainer execution command
+        # @option context [#to_s] :instance_name ("desktop") a name for the
+        #   apptainer instance
```

Comment:
Do we need this as an option to the user?  Do you have a use case for needing something specific vs a hard coded string?

Reply:
I don't think it needs to be a user option, but there does need to be some collision protection against running instances with the same name. A random string would accomplish the same thing.

Reply:
Resolved - removed as user option